### PR TITLE
New Jetpack Admin UI

### PR DESCRIPTION
**Pull request by dereksmart**
**Testing instructions** (also found in the readme.md in this branch):
- Make sure you have git, node, and npm installed and a working WordPress installation.
- `npm run distclean`
- `npm run build` 
- Open /wp-admin/admin.php?page=jetpack in your browser.
- Connect to WordPress.com

**The only blockers for merging will be:** 
- Unable to build 
- Cannot see the admin UI in a CURRENT browser version.  (will fix old IE later)
- Unable to connect or disconnect from WordPress.com 

We will start moving existing issues and PR's over to this repo after this is merged.  

For a pre-built version of this branch, `git checkout jetpack-react-built`

### CODE REVIEW

**Comment by dereksmart** [comment]
if you see `stats.jsx:130Uncaught TypeError: Cannot read property 'views_today' of undefined`, please try to `npm run build` again.

File: _inc/lib/admin-pages/class.jetpack-admin-page.php [hunk 1]
```diff
@@ -63,9 +63,9 @@ function admin_head() {
 	// Render the page with a common top and bottom part, and page specific
 	// content
 	function render() {
-		$this->admin_page_top();
+//		$this->admin_page_top();
```
Comment (gravityrail) [inline _inc/lib/admin-pages/class.jetpack-admin-page.php]: You could probably just remove these lines now, right?

File: _inc/lib/admin-pages/class.jetpack-landing-page.php [hunk 2]
```diff
@@ -286,7 +286,6 @@ function page_admin_scripts() {
 				'modules'           => Jetpack::get_translated_modules( array_values( Jetpack_Admin::init()->get_modules() ) ),
 				'currentVersion'    => JETPACK__VERSION,
 				'ajaxurl'           => admin_url( 'admin-ajax.php' ),
-				'jumpstart_modules' => $this->jumpstart_module_tag( 'Jumpstart' ),
```
Comment (gravityrail) [inline _inc/lib/admin-pages/class.jetpack-landing-page.php]: Any plans to reinstate this functionality?

File: _inc/lib/admin-pages/class.jetpack-admin-page.php (hunk 1 above)
Comment (dereksmart) [inline _inc/lib/admin-pages/class.jetpack-admin-page.php]: Yes, I would like to clean up this code after it's merged.

File: _inc/lib/admin-pages/class.jetpack-landing-page.php (hunk 2 above)
Comment (dereksmart) [inline _inc/lib/admin-pages/class.jetpack-landing-page.php]: Jumpstart functionality? It's being done in `class.jetpack-react-page.php`.  this file is not needed at all

**Comment by ebinnion** [comment]
I built this PR and the UI and routing seemed to work fine. There are several places where the UI needs work. For example: number of comment blocked is in correct, search is not working, etc.

I tested turning the SSO module on/off and it worked as expected.

**Comment by roccotripaldi** [comment]
I was able to download and build no prob.
I was able to run this branch on my site, connect / disconnect from wpcom, and activate / deactive modules.
Well done folks. It's coming along nicely.

**Comment by akirk** [comment]
@dereksmart This PR introduced lots of whitespace (`\n\t\t`) in translatable strings, thus making them untranslated. Please either remove the word wrap within the string or change it to be a concattenation of partial strings. I know the white space doesn't matter in HTML but it matters in translatable strings because they are basically hash lookups which are no longer possible because the strings changed.

**Pull request merged**